bioblend/galaxy/workflows/__init__.py

```diff
@@ -731,6 +731,89 @@ def delete_workflow(self, workflow_id):
         """
         return self._delete(id=workflow_id)
 
+    def refactor_workflow(self, workflow_id, actions, dry_run=False):
+        """
+        Refactor workflow with given actions.
+
+        :type workflow_id: str
+        :param workflow_id: Encoded workflow ID
+
+        :type actions: list of dicts
+        :param actions: Actions to use for refactoring the workflow. The following
+                        actions are supported: update_step_label, update_step_position,
+                        update_output_label, update_name, update_annotation,
+                        update_license, update_creator, update_report, add_step,
+                        add_input, disconnect, connect, fill_defaults, fill_step_defaults,
+                        extract_input, extract_legacy_parameter,
+                        remove_unlabeled_workflow_outputs,
+                        upgrade_subworkflow, upgrade_tool.
+
+          An example value for the ``actions`` argument might be::
+
+            actions = [
+                {"action_type": "add_input", "type": "data", "label": "foo"},
+                {"action_type": "update_step_label", "label": "bar", "step": {"label": "foo"}},
+            ]
+
+        :type dry_run: bool
+        :param dry_run: When true, perform a dry run where the existing
+                        workflow is preserved. A new workflow is
+                        allocated instead.
```

Comment:
```suggestion
        :param dry_run: When true, perform a dry run where the existing
                        workflow is preserved. The refactored workflow
                        is returned in the output of the method, but not saved
                        on the Galaxy server.
```

Reply:
I don't think a dry_run creates a new workflow? @rikeshi please correct me if I'm wrong.

Also, could you add `rtype` for this method please?

Reply:
I was not sure, the code on the Galaxy side did not have it documented. This is what I made of it. You are likely correct.

Yes, I will add it.

Reply:
I checked to make sure, it doesn't create a new workflow :+1: